js/publicshareauth.js

```diff
@@ -0,0 +1,154 @@
+/**
+ *
+ * @copyright Copyright (c) 2018, Daniel Calviño Sánchez (danxuliu@gmail.com)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+(function(OCA) {
+	'use strict';
+
+	OCA.Talk = OCA.Talk || {};
+	OCA.Talk.PublicShareAuth = {
+
+		init: function() {
+			var self = this;
+
+			this.setupRequestPasswordButton();
+			this.setupLayoutForTalkSidebar();
+
+			$('#request-password-button').click(function() {
+				$('.request-password-wrapper + .error-message').hide();
+
+				$('#request-password-button').prop('disabled', 'true');
+
+				$('.request-password-wrapper .icon')
+						.removeClass('icon-confirm-white')
+						.addClass('icon-loading-small-dark');
+
+				self.requestPassword();
+			});
+		},
+
+		setupRequestPasswordButton: function() {
+			// "submit-wrapper" is used to mimic the login button and thus get
+			// automatic colouring of the confirm icon by the Theming app
+			$('main').append('<div id="submit-wrapper" class="request-password-wrapper">' +
```

Comment:
missing the `.` before `main` ?

Reply:
No, it refers to [an element](https://github.com/nextcloud/server/blob/7b848368d52ea84f3ce8e8b8792263de9f394c60/core/templates/layout.guest.php#L40-L42), not to a class.